### ISSUE TITLE
401 Unauthorized: @privateNameSpace/packageName

### ISSUE THREAD

**Issue by rommik** [pnpm/pnpm#2701]
This issue has been reported and closed before here (https://github.com/pnpm/pnpm/issues/618)

It seem to exist in 5.4.0
```
pnpm -v
5.4.0

npm -v
6.14.5
```

.npmrc looks like this

```
@privateNameSpace:registry=https://bin.my-company.com/artifactory/api/npm/npm-local
_authToken=c2Vydmlj(....)DJpR2tBVVEx
always-auth = true
email = email@my-company.com
```
`pnpm add @privateNameSpace/packageName`

I get

```
401 Unauthorized: @privateNameSpace/packageName (via https://bin.my-company.com/artifactory/api/npm/npm-local/@privateNameSpace%2FpackageName)
```

`npm i @privateNameSpace/packageName` 

works as expected

**Comment by zkochan** [comment]
IIRC, the auth token for the non-default registry should be specified by specifying that registry URL. Try this:

```
@privateNameSpace:registry=https://bin.my-company.com/artifactory/api/npm/npm-local
//bin.my-company.com/artifactory/api/npm/:_authToken=c2Vydmlj(....)DJpR2tBVVEx
always-auth = true
email = email@my-company.com
```

**Comment by rommik** [comment]
@zkochan  Thank you for the reply. Unfortunately, I still get the same error. 
Do I need to re-authenticate using pnpm and maybe get a new auth token?

**Comment by zkochan** [comment]
no, the same auth used by npm should work with pnpm. Maybe a related issue: #2687

**Comment by zkochan** [comment]
try pnpm v5.4.9

**Comment by parga** [comment]
@zkochan in my case I get a 403 Forbidden 

.npmrc
```
registry=https://artifactory.company.com:443/artifactory/api/npm/npm-company-release/
//artifactory.company.com:443/artifactory/api/npm/npm-company-release/:_authToken=eyJ2(...)D6g
//artifactory.company.com:443/artifactory/api/npm/npm-company-release/:always-auth=true

@foo:registry=https://artifactory.company.com/artifactory/api/npm/npm-foo-release-local/
@bar:registry=https://artifactory.company.com/artifactory/api/npm/npm-bar-release/
```

error 
```
 ERROR  403 Forbidden: @foo/foo-variables (via https://artifactory.company.com:443/artifactory/api/npm/npm-foo-release-local/@foo%2Ffoo-variables)
```

**Comment by zkochan** [comment]
@parga I don't see auth token provided for the `https://artifactory.company.com/artifactory/api/npm/npm-foo-release-local/` endpoint

**Comment by parga** [comment]
So it’s needed to provide the token to each scope ? The configuration I shared works with yarn and npm, Seems like in npm and yarn they take the auth from the registry, that’s why I was wondering what was missing.

And if it’s needed to add the token to each scope seems repetitive and not aligned with what yarn and npm expect

**Comment by zkochan** [comment]
We need to investigate this case.

I am not sure it is always acceptable. What if the default registry is another domain? Sounds like a dangerous thing to send the auth token everywhere.

**Comment by parga** [comment]
I tried what you said about adding the token to all the scopes, and even that case fails, so seems like something more complex than just a missing piece of config on my side, 

I think for consistency and ease of use would be ideal to just do what npm and yarn do for authorization. 

to give you more context my expectation ( and what has worked with yarn and npm too ) is that the auth token for the registry is in my ~/.npmrc and we define the scopes in the app level .npmrc that way we don't commit any tokens, that support would be ideal

**Comment by zkochan** [comment]
@parga try to install with pnpm v5.5.2 and see the additional information that will be printed in the error message

**Comment by zkochan** [comment]
This seems to be solved in v5.4

Let me know, if no
